File: src/Aspirecafe/Aspirecafe.Shared/Models/Service/Counter/OrderServiceModel.cs
namespace AspireCafe.Shared.Models.Service.Counter
{
    public class OrderServiceModel:ServiceBaseModel
    {
        public OrderHeaderServiceModel Header { get; set; }
        public List<OrderLineItemServiceModel> Lines { get; set; }
        public OrderFooterServiceModel Footer { get; set; }

        public OrderServiceModel()
        {
            Header = new OrderHeaderServiceModel();
            Lines = new List<OrderLineItemServiceModel>();
            Footer = new OrderFooterServiceModel();
        }
    }
}

File: src/Aspirecafe/AspireCafe.UI/src/app/core/models/service/orderservicemodel.ts
export interface OrderServiceModel {
  header: OrderHeaderServiceModel;
  lines: OrderLineItemServiceModel[];
  footer: OrderFooterServiceModel;
}

export interface OrderLineItemServiceModel {
  productId: string;
  quantity: number;
  price: number;
  notes: string | null;
}

export interface OrderHeaderServiceModel {
  orderType: string;
  tableNumber: number | null;
  customerName: string | null;
}

export interface OrderFooterServiceModel {
  subTotal: number;
  tax: number;
  total: number;
  paymentMethod: string;
  paymentStatus: string;
  orderStatus: string;
  notes: string | null;
}
